ammber/BinarySystems.py

```diff
@@ -476,3 +479,307 @@ def get_lower_convex_hull(inputpoints):
+        """
+        Builds discrete systems of phases by evaluating free energies on a range
+        Provide either xdata, or a range and number of points.
+
+        Parameters
+        ----------
+        xdata : list (float)
+            compositions to be sampled
+        xrange : tuple (min, max)
+            range of compositions to be linearly sampled
+        npts : dict
+            number of points to be sampled from in xrange
+
+        Returns
+        -------
+        BinaryDiscretePhase object
+        """
+        discrete_systems = BinaryDiscreteSystem()
+
+        if npts_dict is None:
+            npts_dict = {T: 1001 for T in self.isothermal_systems.keys()}
+        
+        # Add temperatures and corresponding systems to the discrete system
+        for T, system in self.isothermal_systems.items():
+            if T in npts_dict:
+                discrete_systems.add_temperature(T, system.to_discrete(xdata=xdata, xrange=xrange, npts=npts_dict[T]))
+
+        return discrete_systems
+    
+    def fit_parameters(self, phase_name):
+        """
+        Fit the parameters (A, B, D) of a phase.
+
+        Parameters
+        ----------
+        phase_name : str
+            The name of the phase to fit the parameters for.
+        """
+        # Extract kwell, cmin, fmin values for the given phase
+        kwell_list = []
+        cmin_list = []
+        fmin_list = []
+        
+        # Iterate through isothermal systems
+        for temperature in self.isothermal_systems.keys():
+            system = self.isothermal_systems[temperature]
+
+            # Check if the target phase name is contained in any of the keys
+            for ph_name in system.phases.keys():
+                if phase_name in ph_name:
+                    phase_system = system.phases[ph_name]
+                    kwell_list.append((temperature, phase_system.kwell))
+                    cmin_list.append((temperature, phase_system.cmin))
+                    fmin_list.append((temperature, phase_system.fmin))
+                    break  # Found the phase, no need to check further keys
+
+        # Convert lists to dictionaries for kwell, cmin, and fmin
+        kwell_dict = dict(kwell_list)
+        cmin_dict = dict(cmin_list)
+        fmin_dict = dict(fmin_list)
+        
+        # Ensure the lengths of all dicts are sufficient
+        if len(kwell_dict) < 2 or len(cmin_dict) < 2 or len(fmin_dict) < 2:
+            print("Error: Not enough data points to fit one or more parameters.")
+            return
+        
+        # Calculate A, B, D
+        A_dict = {T: k / 2.0 for T, k in kwell_dict.items()}
+        B_dict = {T: -kwell_dict[T] * cmin_dict[T] for T in kwell_dict.keys()}
+        D_dict = {T: 0.5 * kwell_dict[T] * cmin_dict[T] * cmin_dict[T] + fmin_dict[T] for T in kwell_dict.keys()}
+
+        # Fit and plot A, B, D
+        self.fit_and_plot(A_dict, phase_name, 'A', 'blue')
```

Comment:
Let's keep plotting out of this file.  This stuff can go in a separate example like this: https://github.com/wband/AMMBER_python/blob/examples/examples/example_AlCu_eutectic.py

I think we were waiting to merge that in until we had permission to republish the TDB files or something.